Update offsetPath

ecrohr opened the pull request:
Update offsetPath using os.path.split and os.path.join notation for more universal compatibility.

ecrohr committed:
Update groupcat.py

Commit:
typo fix

dnelson commented:
I would typically use a path like `basePath = 'sims.TNG/TNG100-1/output/'`.

In this case, `os.path.split(basePath) == basePath`, correct? Then it seems the logic has changed, i.e. the `../` has been lost?

ecrohr commented:
In the case above, `os.path.split(basePath)` yields `('sims.TNG/TNG100-1', 'output')`. Thereby, `os.path.split(basePath)[0] == 'sims.TNG/TNG100-1'`, and using `os.path.join()` ensures that the `/` between directories is correctly implemented. The resulting offset path would then be `'sims.TNG/TNG100-1/postprocessing/offsets/'` rather than `'sims.TNG/TNG100-1/output/../postprocessing/offsets/'`. These two should be equivalent, but occasionally I have received an OS error using the latter path, which has been resolved in this style.

dnelson commented:
At least in Python 3.12.4:

```
In [1]: from os.path import split

In [2]: split('sims.TNG/TNG100-1/output/')
Out[2]: ('sims.TNG/TNG100-1/output', '')
```

ecrohr commented:
Ahh I see, the difference comes whether the trailing `/` is included in `basePath` or not. When not, then `os.path.split()` works as I wrote, and when yes, then as you wrote. Note that in the current `offsetPath` the result then becomes `sims.TNG/TNG100-1/output//../postprocessing/offsets`, as there is a trailing `/` in `basePath` and a leading `/` is included in the string, which could be the source of the error in some cases. I would still recommend using `os.path.split()` and `os.path.join()` functions for creating the strings for paths and filenames, as this should avoid any issues, but feel free to close this PR without implementing if you wish.

ecrohr commented:
Checking in again about this. The error is somehow related to `hdf5.File()` and read/write permissions when going through the output directory, which tends to be protected, whereas going directly to the postprocessing directory, where a local file structure may not have the same protections, works better. Rather than appending `../postprocessing` to `basePath`, the issue would not arise if instead `output/` were removed from `basePath` before appending the offsets path.

Commit:
add astropy units to snapshot.py

Commit:
add astropy units

Commit:
bookkeeping

dnelson commented:
Hi Eric, I will close this for now: improving the path handling is certainly a good goal, and I agree the use of "../" is probably bad practice. But now I see (strangely) commits into "your" version, about units.

(I would suggest that you build upon `illustris_python`, i.e. import it and go from there, instead of starting to change it. Otherwise your version will become incompatible with this version, for any future updates.)
